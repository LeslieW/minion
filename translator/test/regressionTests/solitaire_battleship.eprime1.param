$  Hints for the following instance of 
$  the solitaire battleship problem:
$
$   1 0 0 0 0 0 0 0 0 0 
$   0 0 0 1 0 0 1 1 1 1 
$   0 0 0 1 0 0 0 0 0 0 
$   0 0 0 1 0 0 0 1 1 1 
$   0 0 0 0 0 0 0 0 0 0 
$   1 1 0 0 0 0 1 0 0 0 
$   0 0 0 0 0 0 1 0 0 0 
$   0 0 0 0 0 0 0 0 0 1 
$   0 1 0 1 0 1 0 0 0 0 
$   0 0 0 1 0 0 0 0 0 0 
$

language ESSENCE' 1.b.a


letting column_tallies be [2,2,0,5,0,1,3,2,2,3]

letting row_tallies be [1,5,1,4,0,3,1,1,3,1]

letting hint be [ [7,7,7,7,7,7,7,7,7,7],
		  [7,7,7,3,7,7,6,7,2,7],
		  [7,7,7,7,7,7,7,7,7,7],
		  [7,7,7,4,7,0,7,6,7,5],
		  [7,7,7,7,7,7,7,7,7,7],		  
		  [7,5,7,0,7,7,3,7,7,7],
		  [7,7,7,7,7,7,7,7,7,7],		  
		  [7,7,7,7,7,7,7,7,7,1],		  
		  [7,1,7,3,7,7,7,7,7,7],
		  [7,7,7,7,7,7,7,7,7,0] ]		  


