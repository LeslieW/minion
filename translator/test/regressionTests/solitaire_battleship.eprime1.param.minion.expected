MINION 3

#This file has been automatically generated by TAILOR v0.2
#bug-reports: andrea@cs.st-and.ac.uk

# amount of common subexpressions used:0
# amount of inferred common subexpressions used:0
# amount of original variables saved (over direct equality):0
# amount of constraints:150
# amount of auxiliary variables: 6

**VARIABLES**
BOOL grid[10,10] 

# auxiliary variables
BOOL aux0 
BOOL aux1 
BOOL aux2 
BOOL aux3 
BOOL aux4 
BOOL aux5 

# aliases

**SEARCH**

PRINT [
grid]

VARORDER [
grid,
aux0,aux1,aux2,aux3,aux4,aux5]


**CONSTRAINTS**

eq(1, grid[1, 6])
eq(0, grid[2, 6])
eq(1, grid[1, 7])
eq(0, grid[0, 6])
eq(0, grid[1, 5])
eq(0, grid[0, 8])
eq(0, grid[0, 7])
eq(0, grid[2, 5])
eq(0, grid[2, 8])
eq(0, grid[2, 7])
eq(0, grid[0, 5])
eq(1, grid[3, 7])
eq(0, grid[4, 7])
eq(1, grid[3, 8])
eq(0, grid[2, 7])
eq(0, grid[3, 6])
eq(0, grid[2, 9])
eq(0, grid[2, 8])
eq(0, grid[4, 6])
eq(0, grid[4, 9])
eq(0, grid[4, 8])
eq(0, grid[2, 6])
eq(1, grid[5, 1])
eq(0, grid[6, 1])
eq(0, grid[5, 2])
eq(1, grid[5, 0])
eq(0, grid[4, 1])
eq(0, grid[6, 0])
eq(0, grid[4, 2])
eq(0, grid[6, 2])
eq(0, grid[4, 0])
eq(1, grid[3, 9])
eq(0, grid[4, 9])
eq(0, grid[2, 9])
eq(1, grid[3, 8])
eq(0, grid[4, 8])
eq(0, grid[4, 7])
eq(0, grid[2, 8])
eq(0, grid[2, 8])
eq(1, grid[3, 3])
eq(0, grid[4, 3])
eq(0, grid[3, 4])
eq(0, grid[3, 2])
eq(1, grid[2, 3])
eq(0, grid[4, 1])
eq(0, grid[4, 2])
eq(0, grid[2, 4])
eq(0, grid[4, 4])
eq(0, grid[1, 2])
eq(0, grid[2, 2])
eq(1, grid[1, 3])
eq(0, grid[1, 4])
eq(1, grid[2, 3])
eq(0, grid[0, 3])
eq(0, grid[1, 2])
eq(0, grid[3, 4])
eq(0, grid[3, 2])
eq(0, grid[2, 2])
eq(0, grid[0, 4])
eq(0, grid[2, 4])
eq(0, grid[0, 2])
eq(1, grid[8, 3])
eq(0, grid[8, 4])
eq(1, grid[9, 3])
eq(0, grid[8, 2])
eq(0, grid[7, 3])
eq(0, grid[9, 2])
eq(0, grid[7, 4])
eq(0, grid[9, 4])
eq(0, grid[7, 2])
eq(1, grid[5, 6])
eq(0, grid[5, 7])
eq(1, grid[6, 6])
eq(0, grid[4, 6])
eq(0, grid[5, 5])
eq(0, grid[7, 7])
eq(0, grid[7, 5])
eq(0, grid[6, 5])
eq(0, grid[4, 7])
eq(0, grid[6, 7])
eq(0, grid[4, 5])
reify(eq(1, grid[0, 8]), aux0)
reify(eq(1, grid[2, 8]), aux1)
reify(sumgeq([aux1,aux0], 2), aux2)
reify(eq(1, grid[1, 7]), aux3)
reify(eq(1, grid[1, 9]), aux4)
reify(sumgeq([aux4,aux3], 2), aux5)
eq(1, grid[1, 8])
eq(0, grid[2, 9])
eq(0, grid[2, 7])
eq(0, grid[0, 9])
eq(0, grid[0, 7])
sumgeq([aux5,aux2], 1)
eq(0, grid[8, 2])
eq(0, grid[9, 1])
eq(0, grid[7, 0])
eq(0, grid[7, 1])
eq(0, grid[7, 2])
eq(0, grid[7, 2])
eq(0, grid[9, 2])
eq(0, grid[7, 0])
eq(0, grid[8, 9])
eq(0, grid[6, 9])
eq(0, grid[6, 8])
eq(0, grid[6, 8])
eq(1, grid[7, 9])
eq(1, grid[8, 1])
eq(0, grid[9, 9])
eq(0, grid[3, 5])
eq(0, grid[5, 3])
watchsumleq([grid[0, 9],grid[1, 9],grid[2, 9],grid[3, 9],grid[4, 9],grid[5, 9],grid[6, 9],grid[7, 9],grid[8, 9],grid[9, 9]], 3)
watchsumgeq([grid[0, 9],grid[1, 9],grid[2, 9],grid[3, 9],grid[4, 9],grid[5, 9],grid[6, 9],grid[7, 9],grid[8, 9],grid[9, 9]], 3)
watchsumleq([grid[0, 8],grid[1, 8],grid[2, 8],grid[3, 8],grid[4, 8],grid[5, 8],grid[6, 8],grid[7, 8],grid[8, 8],grid[9, 8]], 2)
watchsumgeq([grid[0, 8],grid[1, 8],grid[2, 8],grid[3, 8],grid[4, 8],grid[5, 8],grid[6, 8],grid[7, 8],grid[8, 8],grid[9, 8]], 2)
watchsumleq([grid[0, 7],grid[1, 7],grid[2, 7],grid[3, 7],grid[4, 7],grid[5, 7],grid[6, 7],grid[7, 7],grid[8, 7],grid[9, 7]], 2)
watchsumgeq([grid[0, 7],grid[1, 7],grid[2, 7],grid[3, 7],grid[4, 7],grid[5, 7],grid[6, 7],grid[7, 7],grid[8, 7],grid[9, 7]], 2)
watchsumleq([grid[0, 6],grid[1, 6],grid[2, 6],grid[3, 6],grid[4, 6],grid[5, 6],grid[6, 6],grid[7, 6],grid[8, 6],grid[9, 6]], 3)
watchsumgeq([grid[0, 6],grid[1, 6],grid[2, 6],grid[3, 6],grid[4, 6],grid[5, 6],grid[6, 6],grid[7, 6],grid[8, 6],grid[9, 6]], 3)
watchsumleq([grid[0, 5],grid[1, 5],grid[2, 5],grid[3, 5],grid[4, 5],grid[5, 5],grid[6, 5],grid[7, 5],grid[8, 5],grid[9, 5]], 1)
watchsumgeq([grid[0, 5],grid[1, 5],grid[2, 5],grid[3, 5],grid[4, 5],grid[5, 5],grid[6, 5],grid[7, 5],grid[8, 5],grid[9, 5]], 1)
watchsumleq([grid[0, 4],grid[1, 4],grid[2, 4],grid[3, 4],grid[4, 4],grid[5, 4],grid[6, 4],grid[7, 4],grid[8, 4],grid[9, 4]], 0)
watchsumgeq([grid[0, 4],grid[1, 4],grid[2, 4],grid[3, 4],grid[4, 4],grid[5, 4],grid[6, 4],grid[7, 4],grid[8, 4],grid[9, 4]], 0)
watchsumleq([grid[0, 3],grid[1, 3],grid[2, 3],grid[3, 3],grid[4, 3],grid[5, 3],grid[6, 3],grid[7, 3],grid[8, 3],grid[9, 3]], 5)
watchsumgeq([grid[0, 3],grid[1, 3],grid[2, 3],grid[3, 3],grid[4, 3],grid[5, 3],grid[6, 3],grid[7, 3],grid[8, 3],grid[9, 3]], 5)
watchsumleq([grid[0, 2],grid[1, 2],grid[2, 2],grid[3, 2],grid[4, 2],grid[5, 2],grid[6, 2],grid[7, 2],grid[8, 2],grid[9, 2]], 0)
watchsumgeq([grid[0, 2],grid[1, 2],grid[2, 2],grid[3, 2],grid[4, 2],grid[5, 2],grid[6, 2],grid[7, 2],grid[8, 2],grid[9, 2]], 0)
watchsumleq([grid[0, 1],grid[1, 1],grid[2, 1],grid[3, 1],grid[4, 1],grid[5, 1],grid[6, 1],grid[7, 1],grid[8, 1],grid[9, 1]], 2)
watchsumgeq([grid[0, 1],grid[1, 1],grid[2, 1],grid[3, 1],grid[4, 1],grid[5, 1],grid[6, 1],grid[7, 1],grid[8, 1],grid[9, 1]], 2)
watchsumleq([grid[0, 0],grid[1, 0],grid[2, 0],grid[3, 0],grid[4, 0],grid[5, 0],grid[6, 0],grid[7, 0],grid[8, 0],grid[9, 0]], 2)
watchsumgeq([grid[0, 0],grid[1, 0],grid[2, 0],grid[3, 0],grid[4, 0],grid[5, 0],grid[6, 0],grid[7, 0],grid[8, 0],grid[9, 0]], 2)
watchsumleq([grid[9, 0],grid[9, 1],grid[9, 2],grid[9, 3],grid[9, 4],grid[9, 5],grid[9, 6],grid[9, 7],grid[9, 8],grid[9, 9]], 1)
watchsumgeq([grid[9, 0],grid[9, 1],grid[9, 2],grid[9, 3],grid[9, 4],grid[9, 5],grid[9, 6],grid[9, 7],grid[9, 8],grid[9, 9]], 1)
watchsumleq([grid[8, 0],grid[8, 1],grid[8, 2],grid[8, 3],grid[8, 4],grid[8, 5],grid[8, 6],grid[8, 7],grid[8, 8],grid[8, 9]], 3)
watchsumgeq([grid[8, 0],grid[8, 1],grid[8, 2],grid[8, 3],grid[8, 4],grid[8, 5],grid[8, 6],grid[8, 7],grid[8, 8],grid[8, 9]], 3)
watchsumleq([grid[7, 0],grid[7, 1],grid[7, 2],grid[7, 3],grid[7, 4],grid[7, 5],grid[7, 6],grid[7, 7],grid[7, 8],grid[7, 9]], 1)
watchsumgeq([grid[7, 0],grid[7, 1],grid[7, 2],grid[7, 3],grid[7, 4],grid[7, 5],grid[7, 6],grid[7, 7],grid[7, 8],grid[7, 9]], 1)
watchsumleq([grid[6, 0],grid[6, 1],grid[6, 2],grid[6, 3],grid[6, 4],grid[6, 5],grid[6, 6],grid[6, 7],grid[6, 8],grid[6, 9]], 1)
watchsumgeq([grid[6, 0],grid[6, 1],grid[6, 2],grid[6, 3],grid[6, 4],grid[6, 5],grid[6, 6],grid[6, 7],grid[6, 8],grid[6, 9]], 1)
watchsumleq([grid[5, 0],grid[5, 1],grid[5, 2],grid[5, 3],grid[5, 4],grid[5, 5],grid[5, 6],grid[5, 7],grid[5, 8],grid[5, 9]], 3)
watchsumgeq([grid[5, 0],grid[5, 1],grid[5, 2],grid[5, 3],grid[5, 4],grid[5, 5],grid[5, 6],grid[5, 7],grid[5, 8],grid[5, 9]], 3)
watchsumleq([grid[4, 0],grid[4, 1],grid[4, 2],grid[4, 3],grid[4, 4],grid[4, 5],grid[4, 6],grid[4, 7],grid[4, 8],grid[4, 9]], 0)
watchsumgeq([grid[4, 0],grid[4, 1],grid[4, 2],grid[4, 3],grid[4, 4],grid[4, 5],grid[4, 6],grid[4, 7],grid[4, 8],grid[4, 9]], 0)
watchsumleq([grid[3, 0],grid[3, 1],grid[3, 2],grid[3, 3],grid[3, 4],grid[3, 5],grid[3, 6],grid[3, 7],grid[3, 8],grid[3, 9]], 4)
watchsumgeq([grid[3, 0],grid[3, 1],grid[3, 2],grid[3, 3],grid[3, 4],grid[3, 5],grid[3, 6],grid[3, 7],grid[3, 8],grid[3, 9]], 4)
watchsumleq([grid[2, 0],grid[2, 1],grid[2, 2],grid[2, 3],grid[2, 4],grid[2, 5],grid[2, 6],grid[2, 7],grid[2, 8],grid[2, 9]], 1)
watchsumgeq([grid[2, 0],grid[2, 1],grid[2, 2],grid[2, 3],grid[2, 4],grid[2, 5],grid[2, 6],grid[2, 7],grid[2, 8],grid[2, 9]], 1)
watchsumleq([grid[1, 0],grid[1, 1],grid[1, 2],grid[1, 3],grid[1, 4],grid[1, 5],grid[1, 6],grid[1, 7],grid[1, 8],grid[1, 9]], 5)
watchsumgeq([grid[1, 0],grid[1, 1],grid[1, 2],grid[1, 3],grid[1, 4],grid[1, 5],grid[1, 6],grid[1, 7],grid[1, 8],grid[1, 9]], 5)
watchsumleq([grid[0, 0],grid[0, 1],grid[0, 2],grid[0, 3],grid[0, 4],grid[0, 5],grid[0, 6],grid[0, 7],grid[0, 8],grid[0, 9]], 1)
watchsumgeq([grid[0, 0],grid[0, 1],grid[0, 2],grid[0, 3],grid[0, 4],grid[0, 5],grid[0, 6],grid[0, 7],grid[0, 8],grid[0, 9]], 1)

**EOF**
